### ISSUE TITLE
onednn 2.1 build issue on apple silicon

### ISSUE THREAD

**Issue by chenrui333** [oneapi-src/oneDNN#984]
👋 trying to build the latest release, but run into some build issue. The error log is as below:

<details>
<summary>build failure</summary>

```
In file included from /tmp/onednn-20210216-3218-1l8komi/oneDNN-2.1/src/cpu/aarch64/cpu_isa_traits.cpp:23:
In file included from /tmp/onednn-20210216-3218-1l8komi/oneDNN-2.1/src/cpu/aarch64/cpu_isa_traits.hpp:33:
/tmp/onednn-20210216-3218-1l8komi/oneDNN-2.1/src/cpu/aarch64/xbyak_aarch64/xbyak_aarch64_util.h:20:10: fatal error: 'sys/prctl.h' file not found
#include <sys/prctl.h>
         ^~~~~~~~~~~~~
In file included from /tmp/onednn-20210216-3218-1l8komi/oneDNN-2.1/src/cpu/aarch64/injectors/jit_uni_eltwise_injector.cpp:23:
In file included from /tmp/onednn-20210216-3218-1l8komi/oneDNN-2.1/src/cpu/aarch64/injectors/jit_uni_eltwise_injector.hpp:28:
In file included from /tmp/onednn-20210216-3218-1l8komi/oneDNN-2.1/src/cpu/aarch64/injectors/injector_utils.hpp:22:
In file included from /tmp/onednn-20210216-3218-1l8komi/oneDNN-2.1/src/cpu/aarch64/jit_generator.hpp:27:
In file included from /tmp/onednn-20210216-3218-1l8komi/oneDNN-2.1/src/cpu/aarch64/cpu_isa_traits.hpp:33:
/tmp/onednn-20210216-3218-1l8komi/oneDNN-2.1/src/cpu/aarch64/xbyak_aarch64/xbyak_aarch64_util.h:20:10: fatal error: 'sys/prctl.h' file not found
#include <sys/prctl.h>
         ^~~~~~~~~~~~~
In file included from /tmp/onednn-20210216-3218-1l8komi/oneDNN-2.1/src/cpu/aarch64/cpu_barrier.cpp:20:
In file included from /tmp/onednn-20210216-3218-1l8komi/oneDNN-2.1/src/cpu/aarch64/cpu_barrier.hpp:24:
In file included from /tmp/onednn-20210216-3218-1l8komi/oneDNN-2.1/src/cpu/aarch64/jit_generator.hpp:27:
In file included from /tmp/onednn-20210216-3218-1l8komi/oneDNN-2.1/src/cpu/aarch64/cpu_isa_traits.hpp:33:
/tmp/onednn-20210216-3218-1l8komi/oneDNN-2.1/src/cpu/aarch64/xbyak_aarch64/xbyak_aarch64_util.h:20:10: fatal error: 'sys/prctl.h' file not found
#include <sys/prctl.h>
         ^~~~~~~~~~~~~
1 error generated.
make[2]: *** [src/cpu/aarch64/CMakeFiles/dnnl_cpu_aarch64.dir/cpu_isa_traits.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
1 error generated.
make[2]: *** [src/cpu/aarch64/CMakeFiles/dnnl_cpu_aarch64.dir/cpu_barrier.cpp.o] Error 1
```

</details>

Full build log is in here, https://github.com/Homebrew/homebrew-core/runs/1915118962
relates to https://github.com/Homebrew/homebrew-core/pull/71310

**Comment by dzarukin** [comment]
Hi @chenrui333, thank you for reporting the issue.

Hi @kawakami-k, looks like you might be interested in this one. A quick search shows `prctl` is available only on Linux, no MacOS support. I also can't find any limitations or notices saying Arm platform doesn't support MacOS, which means we need to modify our codebase to switch off `xbyak_aarch64` for MacOS completely or try a tool compatible with Linux and MacOS or, maybe, something else.

Thanks.

**Comment by kawakami-k** [comment]
Hi, @chenrui333 , @dzarukin 

Xbyak_aarch64 in oneDNN does not support Apple Silicon, but Mitsunari-san, a member of Xbyak_aarch64 developers, fixed Xbyak_aarch64 for Apple Silicon recently. The latest Xbyak_aarch64 is available from https://github.com/fujitsu/xbyak_aarch64
If we replace Xbyak_aarch64 of oneDNN into the latest version, it may work well.

There is a MAC mini of Apple Silicon in front of me and I want to try it, but no enough time :-(

**Comment by kawakami-k** [comment]
I'm preparing PR which replace Xbyak_aarch64 into the latest version. I think it work well on M1 (Apple Silicon) MAC.

**Comment by kawakami-k** [comment]
HI, @chenrui333, @bayandin

I made a patch for the M1 mac, you can refer to it if you want.
https://github.com/oneapi-src/oneDNN/pull/1001

**Comment by chenrui333** [comment]
let me give that a shot in [this PR](https://github.com/Homebrew/homebrew-core/pull/72003)

**Comment by vpirogov** [comment]
@chenrui333,

The fix is promoted to oneDNN main branch (https://github.com/oneapi-src/oneDNN/pull/1001). Please let us know if this addresses the issue.

**Comment by vpirogov** [comment]
This bug impacts conda-forge packages for v2.1 and v2.2: https://github.com/conda-forge/onednn-feedstock/pull/3.

**Comment by vpirogov** [comment]
Confirmed the issue is fixed in oneDNN v2.1.3 and v2.2.
